src/platform/nm-platform-utils.c

```diff
@@ -395,6 +447,7 @@ nmp_utils_ethtool_set_link_settings (int ifindex,
 	};
 
 	g_return_val_if_fail (ifindex > 0, FALSE);
+	g_return_val_if_fail ((!speed && duplex) || (speed && !duplex), FALSE);
```

Comment:
Why this check? And perhaps it should compare duplex with an enum value, not assume that zero == UNKNOWN.

Reply:
If a mode is specified, both speed and duplex should have a value (otherwise the mode is unspecified). While this could be anyway fine when autoneg is off, because we will keep the missing duplex or speed from the current status this does not work when autoneg is on  and we need to specify a particular mode.
We assume in the logging that the duplex is specified when the speed is.
 So, let's enforce it at the beginning of the function and check to have either both specified or both unset.
Ok, let's check duplex against NM_PLATFORM_LINK_DUPLEX_UNKNOWN